internal/core/command/get.go

```diff
@@ -30,11 +32,23 @@ import (
 func NewGetCommand(
 	device contract.Device,
 	command contract.Command,
-	queryParams string,
 	context context.Context,
 	httpCaller internal.HttpCaller,
-	lc logger.LoggingClient) (Executor, error) {
-	urlResult := device.Service.Addressable.GetBaseURL() + strings.Replace(command.Get.Action.Path, DEVICEIDURLPARAM, device.Id, -1) + "?" + queryParams
+	lc logger.LoggingClient,
+	r *http.Request) (Executor, error) {
+
+	var queryParams string
+	if r != nil {
+		queryParams = r.URL.RawQuery
+	} else {
+		return serviceCommand{}, errors.New("error in request parameters that was passed in")
+	}
```

Comment:
See my previous comment regarding inverting the conditional logic for simplicity, and elimination of side effects.